lib/xgu/xgu.h

```diff
@@ -0,0 +1,937 @@
+    XGU_CULL_FRONT_AND_BACK = NV097_SET_CULL_FACE_V_FRONT_AND_BACK
+} XguCullFace;
+
+typedef enum {
+    XGU_TEXGEN_DISABLE = NV097_SET_TEXGEN_S_DISABLE,
+    XGU_TEXGEN_EYE_LINEAR = NV097_SET_TEXGEN_S_EYE_LINEAR,
+    XGU_TEXGEN_OBJECT_LINEAR = NV097_SET_TEXGEN_S_OBJECT_LINEAR,
+    XGU_TEXGEN_SPHERE_MAP = NV097_SET_TEXGEN_S_SPHERE_MAP,
+    XGU_TEXGEN_REFLECTION_MAP = NV097_SET_TEXGEN_S_REFLECTION_MAP,
+    XGU_TEXGEN_NORMAL_MAP = NV097_SET_TEXGEN_S_NORMAL_MAP
+} XguTexgen;
+
+typedef enum {
+    XGU_SOURCE_TEXTURE = NV097_SET_TEXTURE_FORMAT_BORDER_SOURCE_TEXTURE,
+    XGU_SOURCE_COLOR = NV097_SET_TEXTURE_FORMAT_BORDER_SOURCE_COLOR
+} XguBorderSrc;
+
+typedef enum {
+    XGU_TEXTURE_FORMAT_COLOR_SZ_Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_Y8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_AY8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_AY8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_X1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_X1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A4R4G4B4 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A4R4G4B4,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_R5G6B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_R5G6B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_X8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_X8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT23_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT23_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT45_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT45_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_R5G6B5 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_R5G6B5,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_Y8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8Y8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_AY8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_AY8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_X1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_X1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A4R4G4B4 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A4R4G4B4,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_X8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_X8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8Y8,
+    XGU_TEXTURE_FORMAT_COLOR_LC_IMAGE_CR8YB8CB8YA8 = NV097_SET_TEXTURE_FORMAT_COLOR_LC_IMAGE_CR8YB8CB8YA8,
```

Comment:
not sure what `LC` stands for but we might want to name these differently / add a suffix to mark them as YUV / YUY / YCbCr - I'd suggest to comment them out for now until we figure them out